### PR TITLE
create the cli for baremetalinstance

### PR DESCRIPTION
**Pull request by pabowers**
Create the CLI

For now, I'm using the locally generated python library from the swagger spec, but will update to use the official python SDK library once the python SDK is officially added. The request has been submitted

### CODE REVIEW

File: azext_baremetalinfrastructure/_help.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+
+from knack.help_files import helps
+
+
+helps['baremetalinstance'] = """
+    type: group
+    short-summary: (PREVIEW) Manage Bare Metal Instances.
```
[the remark this answers is not in the document]
Comment (pabowers) [reply azext_baremetalinfrastructure/_help.py:12]: I must have missed that reference

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Microsoft Azure CLI 'azurebaremetal' Command Module for BareMetalInfrastructure
```
[the remark this answers is not in the document]
Comment (pabowers) [reply README.rst:1]: yes it should be baremetalinstance

File: README.md [hunk 3]
```diff
@@ -1,12 +1,77 @@
...
+```
+az extension add --source $LATEST_RELEASE
+```
+
+# Usage
+
+To list all AzureBareMetal instances for the subscription:
+
+```
+az azurebaremetalinstance list
```
[the remark this answers is not in the document]
Comment (pabowers) [reply README.md]: done

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+with open('README.rst', 'r', encoding='utf-8') as f:
+    README = f.read()
+with open('HISTORY.rst', 'r', encoding='utf-8') as f:
+    HISTORY = f.read()
+
+setup(
+    name='azure-baremetal',
+    version=VERSION,
+    description='Additional commands for working with Azure Bare Metal instances.',
```
[the remark this answers is not in the document]
Comment (pabowers) [reply setup.py:37]: done

**Review by AprilZhaoWY — APPROVED** [review]
LGTM

File: README.md [hunk 5]
```diff
@@ -1,7 +1,72 @@
...
+az baremetalinstance delete --resource-group $RESOURCE_GROUP --instance-name $BM_INSTANCE_NAME
+```
+
+To start a specific BareMetal instance:
+
+```
+az baremetalinstance start --resource-group $RESOURCE_GROUP --instance-name $BM_INSTANCE_NAME
+```
+
+To shutdown a specific BareMetal instance:
```
Comment (PakDLiu) [inline README.md:61]: Where is restart?

**Review by PakDLiu — CHANGES_REQUESTED** [review]

**Review by PakDLiu — APPROVED** [review]
Otherwise LGTM

File: README.md (hunk 5 above)
Comment (pabowers) [reply README.md:61]: that was an over sight on my part. Will add now

**Pull request merged**